Adding relevant recipe requirements to atomic butter

PolygonalPerson opened the pull request:
<!-- HOW TO USE: Under each "#### Heading" below, enter information relevant to your pull request.
Leave the headings unless they don't apply to your PR.

Please read carefully and don't delete the comments delimited by "< !--" and "-- >"
Once a pull request is submitted, automatic stylistic and consistency checks will be performed on the PR's changes.
The results of these can be either seen under the "Files changed" section of a PR or in the check's details.

NOTE: Please grant permission for repository maintainers to edit your PR.  It is EXTREMELY common for PRs to be held up due to trivial changes being requested and the author being unavailable to make them. -->

#### Summary
Bugfixes "Adds missing recipe requirments to atomic butter"
<!-- This section should consist of exactly one line, edit the one above.
1. Replace the word "Category" with one of these words: Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N.
2. Replace the text inside the quotes with a brief description of your changes.
Or if you don't want a changelog entry, replace the whole line with just the word "None" (with no quotes).
For more on the meaning of each category, see:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CHANGELOG_GUIDELINES.md
If approved and merged, your summary will be added to the project changelog:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/changelog.txt -->

#### Purpose of change
Allows atomic butter to act as a kind of butter.
<!-- With a few sentences, describe your reasons for making this change.  If it relates to an existing issue, you can link it with a # followed by the GitHub issue number, like #1234.  If your pull request *fully* resolves an issue, include the word "Fix" or "Fixes" before the issue number, like: Fixes #xxxx
If there is no related issue, explain here what issue, feature, or other concern you are addressing.  If this is a bugfix, include steps to reproduce the original bug, so your fix can be verified. -->

#### Describe the solution
Added any_butter & any_butter_or_oil to the requirements, and assigned atomic butter to these, allowing it to be used by any recipe that requires those tags.
<!-- How does the feature work, or how does this fix a bug?  The easier you make your solution to understand, the faster it can get merged. -->

#### Testing
Loaded into CDDA, checked meat confit for the any_butter tag-- worked! Then check with glazed tenderloins for that any_butter_or_oil tag-- worked! Atomic butter was successfully assigned to both tags.
<!-- Describe what steps you took to test that this PR resolved the bug or added the feature, and what tests you performed to make sure it didn't cause any regressions.  Also include testing suggestions for reviewers and maintainers. -->


<!--README: Cataclysm: Dark Days Ahead is released under the Creative Commons Attribution ShareAlike 3.0 license.
The code and content of the game is free to use, modify, and redistribute for any purpose whatsoever.
By contributing to the project you agree to the term of the license and that any contribution you make will also be covered by the same license.
See http://creativecommons.org/licenses/by-sa/3.0/ for details. -->

PolygonalPerson committed:
Update requirements.json

Adds atomic butter to;
any_butter & any_butter_or_oil

Maleclypse commented:
So looking at how the others do it 
```
  {
    "id": "edible_lard",
    "type": "requirement",
    "//": "Lard that is 'safe' for consumption.",
    "extend": { "components": [ [ [ "frost_human_lard", 1 ], [ "alien_lard", 1 ] ] ] }
  },
```
Did you check that regular butter could still be used?   I thought that requirements implicitly extended but I wasn't sure about that.

PolygonalPerson commented:
Good catch! Meat confit seems to work, but all other instances of any_butter or any_butter_or_oil didn't-- atomic butter overwrote it. I'll add extend to each, testing out the change shows that it's playing nice once that's added o7

PolygonalPerson committed:
Update requirements.json to add "extend":

Adds "extend": to Aftershock's requirements.json, to fix the error where atomic butter would overwrite both catagories.
